### ISSUE TITLE
NPM error: npm ERR! code SELF_SIGNED_CERT_IN_CHAIN

### ISSUE THREAD

**Issue by picous** [eclipse/wildwebdeveloper#579]
I am working in a corporation behind firewall. I have no other NPM or Node installed on the workstation. Please advise on a fix to prevent following error:

This is in my eclipse console:

[Error]	[tsserver] npm ERR! code SELF_SIGNED_CERT_IN_CHAIN
npm ERR! errno SELF_SIGNED_CERT_IN_CHAIN
npm ERR! request to https://registry.npmjs.org/types-registry failed, reason: self signed certificate in certificate chain

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\xxxxxxxx\AppData\Roaming\npm-cache\_logs\2020-12-08T22_17_45_263Z-debug.log

Here is the log:

0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   'E:\\Workspace\\cc7Poc202009\\.metadata\\.plugins\\org.eclipse.wildwebdeveloper.embedder.node\\node-v12.16.1-win-x64\\node.exe',
1 verbose cli   'E:\\Workspace\\cc7Poc202009\\.metadata\\.plugins\\org.eclipse.wildwebdeveloper.embedder.node\\node-v12.16.1-win-x64\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   '--ignore-scripts',
1 verbose cli   'types-registry@latest'
1 verbose cli ]
2 info using npm@6.13.4
3 info using node@v12.16.1
4 verbose npm-session 42314a645de8305d
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 silly fetchPackageMetaData error for types-registry@latest request to https://registry.npmjs.org/types-registry failed, reason: self signed certificate in certificate chain
8 timing stage:rollbackFailedOptional Completed in 1ms
9 timing stage:runTopLevelLifecycles Completed in 198ms
10 verbose type system
11 verbose stack FetchError: request to https://registry.npmjs.org/types-registry failed, reason: self signed certificate in certificate chain
11 verbose stack     at ClientRequest.<anonymous> (E:\Workspace\cc7Poc202009\.metadata\.plugins\org.eclipse.wildwebdeveloper.embedder.node\node-v12.16.1-win-x64\node_modules\npm\node_modules\node-fetch-npm\src\index.js:68:14)
11 verbose stack     at ClientRequest.emit (events.js:311:20)
11 verbose stack     at TLSSocket.socketErrorListener (_http_client.js:426:9)
11 verbose stack     at TLSSocket.emit (events.js:311:20)
11 verbose stack     at emitErrorNT (internal/streams/destroy.js:92:8)
11 verbose stack     at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
11 verbose stack     at processTicksAndRejections (internal/process/task_queues.js:84:21)
12 verbose cwd C:\Users\pp51561\AppData\Local\Microsoft\TypeScript\3.9
13 verbose Windows_NT 10.0.17763
14 verbose argv "E:\\Workspace\\cc7Poc202009\\.metadata\\.plugins\\org.eclipse.wildwebdeveloper.embedder.node\\node-v12.16.1-win-x64\\node.exe" "E:\\Workspace\\cc7Poc202009\\.metadata\\.plugins\\org.eclipse.wildwebdeveloper.embedder.node\\node-v12.16.1-win-x64\\node_modules\\npm\\bin\\npm-cli.js" "install" "--ignore-scripts" "types-registry@latest"
15 verbose node v12.16.1
16 verbose npm  v6.13.4
17 error code SELF_SIGNED_CERT_IN_CHAIN
18 error errno SELF_SIGNED_CERT_IN_CHAIN
19 error request to https://registry.npmjs.org/types-registry failed, reason: self signed certificate in certificate chain
20 verbose exit [ 1, true ]

**Comment by vrubezhny** [comment]
Hello, @picous

Have you tried changing `https:` to `http:` in npm registry address, like described here: https://stackoverflow.com/questions/34498736/npm-self-signed-cert-in-chain ?
```
npm config set registry="http://registry.npmjs.org/"
```

**Comment by picous** [comment]
Hi Victor,
as I mentioned, I don't have NPM installed, only as part of the wildwebdeveloper eclipse extension. How do I run the npm config command?

**Comment by vrubezhny** [comment]
@picous I suppose you can do the following ('m not in Win, so I may be mistaken, but you can try fixing the following command according to your actual environment):
- Open CMD (I suppose, you're on Win10?), change directory to your workspace:
```
cd E:\Workspace\cc7Poc202009
```
- To change the npm regstry address:
```
E:\Workspace\cc7Poc202009\.metadata\.plugins\org.eclipse.wildwebdeveloper.embedder.node\node-v12.16.1-win-x64\npm.cmd config set registry="http://registry.npmjs.org/" 
```
- To check what is the actual value of the registry address:
```
E:\Workspace\cc7Poc202009\.metadata\.plugins\org.eclipse.wildwebdeveloper.embedder.node\node-v12.16.1-win-x64\npm.cmd config get registry 
```
HTH 
(Sorry, I'm neither on Win, nor behind a firewall, so I cannot prove if these commands are really must work, but it worth to try)

**Comment by vrubezhny** [comment]
@picous Actually, the npm registry address can be manually changed - it should probably be a file, named like `.npmrc` placed into the root of you project or into your user home directory with the following contents:
```
registry=http://registry.npmjs.org/
```

**Comment by mickaelistria** [comment]
@picous Can you please try to install npm locally and try it? If the result is the same, then we'll have to close this issue as it's not in Wild Web Developer integration specifically.

**Comment by jabby** [comment]
Hello @picous 
I don't know if you tried this, but as I remembered it was helping when I run into in the same problem few years ago. 
https://stackoverflow.com/a/30341389/1947164

**Comment by picous** [comment]
> @picous I suppose you can do the following ('m not in Win, so I may be mistaken, but you can try fixing the following command according to your actual environment):
> 
> * Open CMD (I suppose, you're on Win10?), change directory to your workspace:
> 
> ```
> cd E:\Workspace\cc7Poc202009
> ```
> 
> * To change the npm regstry address:
> 
> ```
> E:\Workspace\cc7Poc202009\.metadata\.plugins\org.eclipse.wildwebdeveloper.embedder.node\node-v12.16.1-win-x64\npm.cmd config set registry="http://registry.npmjs.org/" 
> ```
> 
> * To check what is the actual value of the registry address:
> 
> ```
> E:\Workspace\cc7Poc202009\.metadata\.plugins\org.eclipse.wildwebdeveloper.embedder.node\node-v12.16.1-win-x64\npm.cmd config get registry 
> ```
> 
> HTH
> (Sorry, I'm neither on Win, nor behind a firewall, so I cannot prove if these commands are really must work, but it worth to try)

Thanks Victor, that worked just fine. Only thing is it will have to be done in every workspace...

Now I get different NPM error message - can you please advise?

[Error]	[tsserver] npm notice created a lockfile as package-lock.json. You should commit this file.

**Comment by vrubezhny** [comment]
@picous I glad I could help you.

On package-lock.json - it is normal result of executing of `npm` - here is a good explanation of this file - https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json. 
In short, depending on your the requirements of  your project validation rules. used source code management system etc. you probably need to treat this file as a source file and add commit it into your source code management system (git, cvs, etc. - whatever you use). 
Any change in versions and/or list of modules (made by you or by downloading a new version of any module) will lead into an according change to this file (after build) - so, you'll have to commit its new version again.
Basically it's a part of your project's source code.

**Comment by picous** [comment]
Thanks for the answer.

**Comment by jabby** [comment]
@picous can we close this issue now?

**Comment by picous** [comment]
Yes, thank you

**Comment by Alok07022023** [comment]
This is also work for me
//Turns off ssl
npm set strict-ssl false
//Installs node packages
npm ci
//Turns ssl back on for future protection
npm set strict-ssl true
